server_stream_handler.go

```diff
@@ -0,0 +1,225 @@
+package psrpc
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/livekit/psrpc/internal"
+)
+
+type streamRPCHandlerImpl[RecvType, SendType proto.Message] struct {
```

Comment:
Let's combine this with rpcHandlerImpl, it looks like 80+% of the code is the same

Reply:
it is but i thought it would be easier to review if i implemented the feature and refactored it in separate commits.